apps/game_backend/lib/game_backend/matches.ex

```diff
@@ -117,23 +104,42 @@ defmodule GameBackend.Matches do
   ####################
   #      Helpers     #
   ####################
+  defp match_prestige_reward(nil, position, rewards) do
+    match_prestige_reward(%{amount: 0}, position, rewards)
+  end
 
-  ## TODO: Properly pre-process `currencies_config` so the keys are integers and we don't need conversion
-  ##    https://github.com/lambdaclass/mirra_backend/issues/601
-  def get_amount_of_trophies_to_modify(current_trophies, position, currencies_config) when is_integer(position) do
-    get_amount_of_trophies_to_modify(current_trophies, to_string(position), currencies_config)
+  defp match_prestige_reward(prestige, position, rewards) do
+    Map.get(rewards, position)
+    |> Enum.find(fn %{min: minp, max: maxp} -> prestige.amount in minp..maxp end)
```

Comment:
I'm a bit concerned about the complexity here, aren't we doing a lot of extra comparisons here? for example, if we're level 999 we'll do 999 comparisons in the worst-case scenario? if we do `prestige.amount > minp and prestige.amount <= maxp` we'll only do 4 comparisons in the worst-case scenario. Correct me if I'm wrong

Reply:
TL;DR `in` for a range gets converted to that same expression you wrote

Ranges are not actually a list of numbers, they are a struct that represent a start, end, and step ([Representation](https://hexdocs.pm/elixir/Range.html#module-representation)), this means that is can have its own implementations of protocols, in particular `Enumerable`

`in` is a macro and gets translated to `Enum.member?/2` ([docs](https://hexdocs.pm/elixir/Kernel.html#in/2))

So, if you look at the code for [Range.member?/2](https://github.com/elixir-lang/elixir/blob/v1.16.3/lib/elixir/lib/range.ex#L545) you will see it is essentially the same check that you mention (plus something to account for the step)

Reply:
Ooohhh that's good to know! thanks for taking the time to write this down :)